src/controllers/restful/PaymentController.ts

```diff
@@ -4,223 +4,313 @@
+    if (isLeft(errorOrInodoVerificaRPTOutput)) {
+      const error = errorOrInodoVerificaRPTOutput.value;
       return ResponseErrorInternal(
-        `PagoPa Server communication error: ${error.message}`
+        `PagoPA Server communication error: ${error.message}`
       );
     }
-    const paymentCheckPagoPaResponse = errorOrPaymentCheckPagoPaResponse.value;
+    const iNodoVerificaRPTOutput = errorOrInodoVerificaRPTOutput.value;
 
-    // Check PagoPa response content.
+    // Check PagoPA response content.
     // If it contains an error, an HTTP error will be provided to BackendApp
-    if (
-      paymentCheckPagoPaResponse.nodoVerificaRPTRisposta.esito ===
-      PPTPortTypes.Esito.KO
-    ) {
+    if (iNodoVerificaRPTOutput.esito === PPTPortTypes.Esito.KO) {
       return ResponseErrorInternal("Error during payment check: esito === KO");
     }
 
-    // Convert the output provided by PagoPa (SOAP response)
+    // Convert the output provided by PagoPA (SOAP response)
     // to a BackendApp response (RESTful response), mapping the result information.
     // Send a response to BackendApp
-    return PaymentsConverter.getPaymentsCheckResponse(
-      paymentCheckPagoPaResponse,
+    return PaymentsConverter.getPaymentRequestsGetResponse(
+      iNodoVerificaRPTOutput,
       codiceContestoPagamento
     ).fold<
-      IResponseErrorValidation | IResponseSuccessJson<PaymentsCheckResponse>
+      | IResponseErrorValidation
+      | IResponseSuccessJson<PaymentRequestsGetResponse>
     >(
-      ResponseErrorFromValidationErrors(PaymentsCheckResponse),
+      ResponseErrorFromValidationErrors(PaymentRequestsGetResponse),
       ResponseSuccessJson
     );
   };
 }
 
 /**
  * This controller will be invoked by BackendApp.
- * It's necessary to start the payment process for a specific qrcode (payment).
- * It will require the payment lock to PagoPa (AttivaRPT service) to avoid concurrency problems.
+ * It's necessary to start the payment process for a specific payment.
+ * It will require the payment lock to PagoPA (AttivaRPT service) to avoid concurrency problems.
  * This request result will confirm the taking charge about the payment lock request.
- * If success, it will be necessary to wait an async response from PagoPa.
+ * If success, it will be necessary to wait an async response from PagoPA.
  * @param {express.Request} req - The RESTful request
- * @param {PagoPaConfig} pagoPaConfig - Configuration about PagoPa WS to contact
- * @return {Promise<PaymentCtrlResponse<PaymentsActivationResponse>>} The response content to send to applicant
+ * @param {PagoPAConfig} pagoPAConfig - Configuration about PagoPA WS to contact
+ * @return {Promise<PaymentCtrlResponse<PaymentActivationsPostRequest>>} The response content to send to applicant
  */
-export function activatePaymentToPagoPa(
-  pagoPaConfig: PagoPaConfig,
-  attivaRPTPagoPaClient: pagoPaSoapClient.PagamentiTelematiciPspNodoAsyncClient
+export function paymentActivationsPost(
```

Comment:
can we name these methods like the operationId in swagger specs (ie. `activatePayment`) ?